lib/default-model.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const url = require('url');
+const _ = require('lodash');
+const Model = require('hof-model');
+const defaults = require('./defaults');
+
+module.exports = class PostcodesModel extends Model {
+  constructor(options) {
+    super(options);
+    this.options = Object.assign({}, defaults.POSTCODE_API, options);
+  }
+
+  fetch(postcode) {
+    return new Promise((resolve, reject) => {
+      if (_.startsWith(postcode.toUpperCase(), 'BT')) {
+        const err = new Error('Postcode not supported');
+        err.status = 501;
+        return reject(err);
+      }
+      const attributes = {
+        url: `${this.options.hostname}/${this.options.paths.lookup}`,
+        query: {
+          postcode
+        }
+      };
+      const reqConf = url.parse(this.url(attributes));
+
+      reqConf.method = 'GET';
+      reqConf.headers = {
+        Authorization: this.options.authorization || ''
+      };
+      this.request(reqConf, (err, data) => {
+        if (err) {
+          return reject(err);
+        }
+        return resolve(data);
+      });
+    });
```

Comment:
fair, this is mostly lifted from first implementation - I'll have a better look at hof-model